src/zabapgit_gui_router.prog.abap

```diff
@@ -289,9 +296,10 @@ CLASS lcl_gui_router IMPLEMENTATION.
 
     CREATE OBJECT lo_page
       EXPORTING
-        iv_key    = lv_key
-        is_file   = ls_file
-        is_object = ls_object.
+        iv_key           = lv_key
+        is_file          = ls_file
+        is_object        = ls_object
+        iv_supress_stage = boolc( iv_prev_page = 'PAGE_STAGE' ).
```

Comment:
Generally agree with both points. But it is not that clear how to do it. 
There is also one use case for DB_DISPLAY/DB_EDIT.

Thoughts:
- for DB pages EDIT and DISPLAY can be merged in one class - I think it would be reasonable. Then the issue go completely.
- for DIFF/STAGE ... I don't know. I guess there should be some common context accessible by any page. But this raises completely new challenges.